File: cpp/3rd_Week/Praktikum/latihanpraktikum3_cpp/Disk.cpp
#include <iostream>
#include <string>

using namespace std;

class Disk
{
private:
    /* data */
    string type, diskCapacity;
    long diskPrice;
public:
// contructor
    Disk() {}
    Disk(string type, string capacity, long diskPrice){
        this->type = type;
        this->diskCapacity = capacity;
        this->diskPrice = diskPrice;
    }

    // set and get atribut
    void setType(string t){ this->type = t; }
    void setDiskCapacity(string c){ this->diskCapacity = c; }
    void setDiskPrice(long p){ this->diskPrice = p; }
    
    string getType(){ return this->type; }
    string getDiskCapacity(){ return this->diskCapacity; }
    long getDiskPrice(){ return this->diskPrice; }

    //method print
    void printDisk(){
        cout << "-Data Disk" << endl;
        cout << " Type          : " << this->type << endl;
        cout << " Disk Capicity : " << this->diskCapacity << endl;
        cout << " Disk Price    : " << this->diskPrice << endl;
        cout << "-----------------------------------" << endl;
    }
    ~Disk() {}
};

File: php/3rd_week/latihanpraktikum3_php/Disk.php
<?php

class Disk{
    // implementasi class Disk

    // atribut
    private $type = "";
    private $diskCapacity = "";
    private $diskPrice = 0;

    //constructor
    function __construct()
    {
    }

    //set and get atribut
    function setType($type){
        $this->type = $type;
    }

    function setDiskCapacity($capacity){
        $this->diskCapacity = $capacity;
    }

    function setDiskPrice($price){
        $this->diskPrice = $price;
    }

    function getType(){
        return $this->type;
    }

    function getDiskCapacity(){
        return $this->diskCapacity;
    }

    function getDiskPrice(){
        return $this->diskPrice;
    }

    // method print disk
    function printDisk(){
        echo "-Data Disk". "<br/>";
        echo " Type : ". $this->type. "<br/>";
        echo " Disk Capacity : ". $this->diskCapacity. "<br/>";
        echo " Harga Disk : ". $this->diskPrice. "<br/>";
        echo "--------------------------------------------". "<br/>";
    }

    //destructor
    function __destruct()
    {
    }
}
?>